datafusion/functions/src/string/uuid.rs

```diff
@@ -87,7 +88,13 @@ impl ScalarUDFImpl for UuidFunc {
         if !args.is_empty() {
             return internal_err!("{} function does not accept arguments", self.name());
         }
-        let values = std::iter::repeat_with(|| Uuid::new_v4().to_string()).take(num_rows);
+
+        // Generate random u128 values
+        let mut rng = rand::thread_rng();
+        let mut randoms = vec![0u128; num_rows];
+        rng.fill(&mut randoms[..]);
+
+        let values = randoms.into_iter().map(|x| Uuid::from_u128(x).to_string());
```

Comment:
it would be nice to avoid a `to_string` here as well

Reply:
Resolved in https://github.com/apache/datafusion/pull/14675/commits/f4f87c829dd8ffefff4f8383cccc05222cc40e51

Reply:
It doubled performance :)